sdk/go/x/auto/git.go

```diff
@@ -21,11 +21,57 @@ import (
 	"github.com/pkg/errors"
 	git "gopkg.in/src-d/go-git.v4"
 	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/ssh"
 )
 
 func setupGitRepo(ctx context.Context, workDir string, repoArgs *GitRepo) (string, error) {
+	cloneOptions := &git.CloneOptions{
+		URL: repoArgs.URL,
+	}
+
+	if repoArgs.Auth != nil {
+
+		authDetails := repoArgs.Auth
```

Comment:
nit: we set `authDetails` here but only use it in the mutual exclusive checks. we should make use of this var in the rest of the `if` statements e.g. L45, L47, L65

Reply:
Should be taken care of now!